megatron/core/inference/sampling_params.py

```diff
@@ -50,7 +50,7 @@ def _sync_prompt_logprobs_fields(self):
             assert (
                 self.skip_prompt_log_probs
             ), "return_prompt_top_n_logprobs requires skip_prompt_log_probs to be False"
-        if self.top_n_logprobs > 0:
+        if self.return_log_probs:
```

Comment:
Wouldn't this change make us compute top-n logprobs even if we only want the actual selected token logprobs?